python/pyarrow/_dataset_parquet.pyx

```diff
@@ -606,15 +601,12 @@ cdef class ParquetFragmentScanOptions(FragmentScanOptions):
 
     def __init__(self, bint use_buffered_stream=False,
                  buffer_size=8192,
-                 bint pre_buffer=False,
-                 bint enable_parallel_column_conversion=False):
+                 bint pre_buffer=False):
```

Comment:
Should we accept the parameter, but ignore it/raise a DeprecationWarning? (Change the default value to None, then raise a warning if it's not None). Same goes for the getter/setter below, I suppose.

Reply:
I can.  Though the documentation around the parameter clearly marked it as experimental with a note that it will be removed in the future.

Reply:
Hmm, that's fair. We can just remove it then.

Reply:
@jorisvandenbossche any opinion on whether we should deprecate or remove this flag in datasets?

Reply:
Yeah, I think it is fine to just remove it altogether

Reply:
I'll merge this then, thanks Weston!